core/shared/src/test/scala/fs2/StreamSpec.scala

```diff
@@ -3786,4 +3787,14 @@ class StreamSpec extends Fs2Spec {
       }
     }
   }
+
+  "withTimeout" - {
+    "timeout never-ending stream" in {
```

Comment:
Would you mind also testing composability of timeouts? i.e. if you do `(s1.timeout(d1) ++ s2).timeout(d2)`, what happens depending on the times of `s1`, `s2`, `d1 > d2` (if d2 > d1, s1 might make it in time but s2 might not, so we still want to timeout, etc.)